Shared/Supporting Files/Extensions/View+Sheet.swift

```diff
@@ -286,44 +286,45 @@ private struct Sheet<Content>: UIViewRepresentable where Content: View {
+        
+        /// A Boolean value indicating whether the hosting controller is being dismissed.
+        let isHostBeingDismissed = model.hostingController.isBeingDismissed
+        
         // Ensures that the device's layout is such that a sheet should be presented and
         // the hosting controller's sheet presentation controller exists.
         guard isSheetLayout,
               let sheet = model.hostingController.sheetPresentationController else {
             // Dismisses the sheet if it is being presented and a popover should
             // be presented instead.
-            if isPresentedControllerHostingType && !model.hostingController.isBeingDismissed {
+            if isPresentedControllerHostingType && !isHostBeingDismissed {
                 rootViewController.dismiss(animated: false)
             }
             return
         }
         
-        /// A Boolean value indicating whether the sheet was already presenting.
-        let wasPresenting = rootViewController.presentedViewController != nil
-        
-        /// A Boolean value indicating whether the hosting controller is being dismissed.
-        let isHostBeingDismissed = model.hostingController.isBeingDismissed
-        
-        /// A Boolean value indicating whether the presented view controller is an alert controller.
-        let isPresentedControllerAlertType = rootViewController.presentedViewController is UIAlertController
-        
         if isPresented && !wasPresenting {
             // Sets the sheet presentation controller's delegate.
             sheet.delegate = context.coordinator
             // Configures the hosting controller's sheet presentation controller.
             configureSheetPresentationController(sheet)
             // Presents the hosting controller.
-            rootViewController.present(model.hostingController, animated: model.isTransitioningFromPopover ? false : true)
+            rootViewController.present(model.hostingController, animated: !model.isTransitioningFromPopover)
         } else if !isPresented && wasPresenting && !isHostBeingDismissed && !isPresentedControllerAlertType {
             // Dismisses the view controller presented by the root view controller
             // if 'isPresented' is false, but was presenting before (popover), is
             // not currently being dismissed, and is not an alert.
-            rootViewController.dismiss(animated: isPresentedControllerHostingType ? true : false)
+            rootViewController.dismiss(animated: isPresentedControllerHostingType)
             model.isTransitioningFromPopover = !isPresentedControllerHostingType
         } else if isHostBeingDismissed {
             // Sets 'isPresented' to false when the hosting controller is being dismissed.
-            isPresented = false
-            model.isTransitioningFromPopover = !isPresentedControllerHostingType
+            Task {
+                isPresented = false
+            }
```

Comment:
This is great!

Reply:
The `Task` should be used to execute async code but `isPresented` is not async so what are we trying to do here?

Reply:
The `Task` is used to updating State variables without modifying them while the view is updating otherwise you get: `Modifying state during view update, this will cause undefined behavior.`

Reply:
I don't think this is a right use of the `Task`. The warning comes up because the value is updated while the view is being updated. Should it happen after it finished updating?

Reply:
I am not sure how I would do that. Do you have any suggestions?

Reply:
I agree that it is not the right use of `Task` here. It seems that by wrapping the line in the block, the timing is slightly delayed so that the sheet can be dismissed prior to  `isPresented` is set to `false`.

In the original PR there was a different issue and eventually a `Task` block was used to work around that bug. https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/39#discussion_r933540923 You can find the task block in the condition branch a few lines below.

I'm OK if the threading problem can be mitigated by using a task block albeit it is not a right use - after all, this extension will be removed when we move to iOS 16.

Reply:
If there is no other way then I would suggest using the `DispatchQueue.main.async` instead of `Task`.

```swift
        } else if hostingControllerIsBeingDismissed {
            // Sets 'isPresented' to false when the hosting controller is being dismissed.
            DispatchQueue.main.async {
                isPresented = false
            }
        } else if wasPresenting {
            // Updates the sheet presentation controller and the root view of the hosting
            // controller if the sheet was already presenting.
            configureSheetPresentationController(sheet)
            DispatchQueue.main.async {
                model.hostingController.rootView = content
            }
        }
```

Reply:
If at all possible, GCD and modern Swift concurrency should not be mixed. [`MainActor.run(resultType:body:)`](https://developer.apple.com/documentation/swift/mainactor/run(resulttype:body:)) should be used instead of `DispatchQueue.main.async { ... }`.

Reply:
The above code is part of `func updateUIView(_ uiView: UIView, context: Context)` which is a synchronous function and use of `MainActor.run` requires using `await`. So it can't be used without using `Task` and we are trying to avoid using the `Task`.

Reply:
I don't think using a Task is a problem in this case. Or at least I don't see why you would avoid it here. It's basically has the same cons that dispatch async has.

Reply:
I'm ok if we want to continue using `Task` here. My understanding is that `Task` should be used to execute `async` code from a `sync` context, not to convert the `sync` code to `async`.